### ISSUE TITLE
not all scripts are getting a nonce when using the vite plugin

### ISSUE THREAD

**Issue by mcansh** [remix-run/remix#8010]
### What version of Remix are you using?

0.0.0-nightly-8eb9621-20231114

### Are all your remix dependencies & dev-dependencies using the same version?

- [X] Yes

### Steps to Reproduce

use the vite plugin and add a CSP with a nonce for `script-src`

```ts
script-src 'self' 'nonce-c8rIrQN+BNX9gkm49Qcn3A=='
```

see that the nonce is not applied to the following script 
```html
<script>
  window.__remixLiveReloadEnabled = true
</script>
```

### Expected Behavior

nonce is added to the script tag

```html
<script nonce="c8rIrQN+BNX9gkm49Qcn3A==">
  window.__remixLiveReloadEnabled = true
</script>
```

### Actual Behavior

nonce is not added to the script tag

```html
<script>
  window.__remixLiveReloadEnabled = true
</script>
```

**Comment by markdalgleish** [comment]
Should be fixed by https://github.com/remix-run/remix/pull/8014. This will be in the next nightly if you want to try it out and report back.

**Comment by mcansh** [comment]
> Should be fixed by #8014. This will be in the next nightly if you want to try it out and report back.

can confirm, that fixed it 👍
